src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts

```diff
@@ -26,6 +24,13 @@ import { IDecoration, Terminal } from 'xterm';
 // Importing types is safe in any layer
 // eslint-disable-next-line local/code-import-patterns
 import type { ITerminalAddon } from 'xterm-headless';
+import { CodeActionWidget } from 'vs/editor/contrib/codeAction/browser/codeActionWidget';
+import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { CodeActionKind, CodeActionTrigger, CodeActionTriggerSource } from 'vs/editor/contrib/codeAction/browser/types';
+import { CodeActionTriggerType } from 'vs/editor/common/languages';
+import { CodeActionItem } from 'vs/editor/contrib/codeAction/browser/codeAction';
+import { CancellationToken } from 'vs/base/common/cancellation';
+import { ProviderResult } from 'vs/workbench/services/search/common/searchExtTypes';
```

Comment:
The bigger part of adopting this item will be making the code action widget more generic as we should not be depending on the editor or editor-specific concepts (eg. `CodeAction`) in this way. I would expect a new base `CodeActionWidget.ts` in `vs/base/browser/ui/codeAction/codeActionWidget.ts`  which pulls all the common parts out and then `vs/editor/contrib/codeAction/browser/codeActionWidget.ts` would depend upon the base widget.

This is similar to how hoverWidget.ts in base only has dependencies in base and it's used in both workbench and editor.

cc @mjbvz

Reply:
Yep sounds good and wanted to just see how it looked in the terminal first in case there were any unique considerations we should take into account